packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,175 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import FormLabel from '../FormLabel/FormLabel';
+import PropTypes from 'prop-types';
+import React from 'react';
+import noop from 'nooop';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * An `AutocompleteField` component renders a typeahead input field, form label, and a clear input link.
+ */
+export class AutocompleteField extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      disabled,
+      fieldRef,
+      items,
+      itemToString,
+      labelHint,
+      labelText,
+      onChange,
+      onStateChange,
+      ...fieldProps
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        onStateChange={onStateChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-display--flex ds-u-flex-wrap--wrap ds-c-autocomplete">
```

Comment:
Were these flexbox classes leftovers from a previous implementation of this? Doesn't seem to be anything currently in here that requires this (unless I'm missing something).

Reply:
The Flexbox classes were used to force the layout I wanted, with the clear link always to the lower right. I could try floats and clearfix on the parent div, if there's a strong feeling to move in that direction.

Reply:
Ah ok, got it. I was just missing where it was being used.

Reply:
I opted to go back to a simple float, and clearfix on the containing div. Will be included in the next PR push.